lib-js/compiler.ts

```diff
@@ -13,40 +13,47 @@ export interface Stdio {
 /**
  * A wrapper around a process that can compile Sass files.
  */
-export interface Compiler {
+export abstract class Compiler {
   /**
    * Run the compiler with the given args, at the path given as the cwd.
    */
-  compile(path: string, args: string[]): Promise<Stdio>
+  abstract compile(path: string, args: string[]): Promise<Stdio>
+
+  /**
+   * Shutdowns the compiler in case it was long-running
+   */
+  shutdown(): void {}
 }
 
-/**
- * Returns a sass compiler that runs the given command.
- */
-export function executableCompiler(
-  command: string,
-  initArgs: string[] = []
-): Compiler {
-  return {
-    async compile(path, args) {
-      const { error, stdout, stderr, status } = child_process.spawnSync(
-        command,
-        [...initArgs, ...args],
-        {
-          cwd: path,
-          encoding: "utf-8",
-          stdio: ["ignore", "pipe", "pipe"],
-        }
-      )
-      if (error) {
-        throw new Error(`Failed to run executable compiler: ${error}`)
+export class ExecutableCompiler extends Compiler {
+  private readonly command: string
+  private readonly initArgs: string[]
+
+  constructor(command: string, initArgs: string[] = []) {
```

Comment:
TS lets you instantiate the members here

```
constructor(private readonly command: string, private readonly initArgs: string[] = [])
```

which lets us remove line 29, 30, 34, and 35. Same goes for `MockCompiler`

Reply:
oh, good to know. I didn't know that.